kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestFileSystemWriters.java

```diff
@@ -52,35 +63,74 @@ public void tearDown() throws IOException {
   }
 
   @Test
+  public void testBasicWrite() throws IOException {
+    init(fsWriter);
+
+    FileStatus[] stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should contain no visible files", 0, stats.length);
+    stats = fs.listStatus(testDirectory);
+    Assert.assertEquals("Should contain a hidden file", 1, stats.length);
+
+    List<Record> written = Lists.newArrayList();
+    for (long i = 0; i < 10000; i += 1) {
+      Record record = record(i, "test-" + i);
+      fsWriter.write(record);
+      written.add(record);
+    }
+
+    stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should contain no visible files", 0, stats.length);
+    stats = fs.listStatus(testDirectory);
+    Assert.assertEquals("Should contain a hidden file", 1, stats.length);
+
+    fsWriter.close();
+
+    stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should contain a visible data file", 1, stats.length);
+
+    DatasetReader<Record> reader = newReader(stats[0].getPath(), TEST_SCHEMA);
+    Assert.assertEquals("Should match written records",
+        written, Lists.newArrayList((Iterator) init(reader)));
+  }
+
+  @Test
   public void testDiscardEmptyFiles() throws IOException {
     init(fsWriter);
     fsWriter.close();
-    Assert.assertEquals("Should not contain any files", 0,
-        ImmutableList.copyOf(fs.listStatus(testDirectory)).size());
+
+    FileStatus[] stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should not contain any files", 0, stats.length);
   }
 
   @Test
-  public void testWrite() throws IOException {
```

Comment:
This test used AvroAppender directly, even when testing Parquet. I've updated it to test Parquet and Avro, as testBasicWrites